test/parallel/market.rpc.check.test.ts

```diff
@@ -0,0 +1,473 @@
+/*
+ *
+ * @group marketSwap
+ */
+
+import { jest } from "@jest/globals";
+import { ApiPromise } from "@polkadot/api";
+import { getApi, initApi } from "../../utils/api";
+import { AssetWallet, User } from "../../utils/User";
+import { BN } from "ethereumjs-util";
+import { getSudoUser, setupApi, setupUsers } from "../../utils/setup";
+import {
+  calculateBuyPriceByMarket,
+  calculateBuyPriceWithImpact,
+  calculateSellPriceByMarket,
+  calculateSellPriceWithImpact,
+  getFeeLockMetadata,
+} from "../../utils/feeLockHelper";
+import { stringToBN } from "../../utils/utils";
+import { Sudo } from "../../utils/sudo";
+import { Assets } from "../../utils/Assets";
+import {
+  calculateExpectedLiquidityMinted,
+  getBurnAmount,
+  getPoolId,
+  getPoolsForTrading,
+  getTradeableTokens,
+  Market,
+} from "../../utils/market";
+import {
+  calculate_buy_price_rpc,
+  calculate_sell_price_rpc,
+  getBalanceOfPool,
+  getTokensAccountInfo,
+  updateFeeLockMetadata,
+} from "../../utils/tx";
+import { signTx } from "gasp-sdk";
+import {
+  ExtrinsicResult,
+  filterAndStringifyFirstEvent,
+} from "../../utils/eventListeners";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(1500000);
+process.env.NODE_ENV = "test";
+
+let testUser: User;
+let sudo: User;
+
+let api: ApiPromise;
+let threshold: BN;
+let liqId: BN;
+let priceCalculationRetro: BN;
+
+let firstCurrency: BN;
+let secondCurrency: BN;
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+
+  api = getApi();
+  await setupApi();
+  setupUsers();
+
+  const meta = await getFeeLockMetadata();
+  threshold = stringToBN(
+    JSON.parse(JSON.stringify(meta)).swapValueThreshold.toString(),
+  );
```

Comment:
these tests do not require thresholds, do they? Lets try to avoid some flakiness by setting here an arbitrary amount instead of this threshold from node.

Reply:
Changed